tests/test_git.py

```
@@ -0,0 +1,520 @@
+        assert "token123" not in result
+        assert "***" in result
+
+
+class TestGitRepositoryGenerateUrl:
+    """Tests for GitRepository URL generation."""
+
+    @patch("helm_charts_updater.git.config")
+    @patch("helm_charts_updater.git.Repo")
+    @patch("os.path.exists")
+    def test_generate_repo_url(
+        self, mock_exists: MagicMock, mock_repo: MagicMock, mock_config: MagicMock
```

Ruff (0.14.13)

Unused method argument: `mock_repo`

(ARG002)